src/components/structures/MessagePanel.js

```diff
@@ -471,13 +471,19 @@ module.exports = React.createClass({
         return ret;
     },
 
-    _wantsDateSeparator: function(prevEvent, nextEventTs) {
+    _wantsDateSeparator: function(prevEvent, nextEvent) {
         if (prevEvent == null) {
             // first event in the panel: depends if we could back-paginate from
             // here.
             return !this.props.suppressFirstDateSeparator;
         }
-        return Math.floor(prevEvent.getTs() / MILLIS_IN_DAY) !== Math.floor(nextEventTs / MILLIS_IN_DAY);
+        // Return early for events that are > 24h apart
+        if (Math.abs(prevEvent.getTs() - nextEvent.getTs()) > MILLIS_IN_DAY) {
+            return true;
+        }
+
+        // Compare weekdays
+        return prevEvent.getDate().getDay() !== nextEvent.getDate().getDay();
```

Comment:
> Why does it feel odd? 

Dunno really. Just that day-of-week doesn't often get used in calculations, I guess. But it's fine. Leave it as is.